apps/authentication-service/src/resources/clients/clientsService.ts

```diff
@@ -0,0 +1,95 @@
+import { Client, ClientCreate, ResponseError } from "auth0";
+
+import envs from "@/config";
+import { getAccessToken } from "@/shared/auth/authService";
+import { parseResponse } from "@/shared/utils";
+import {
+  getUserClientId,
+  getUserIdFromToken,
+  setUserClientId,
+} from "@/shared/users/usersService";
+
+const authDomain = envs.AUTH0_DOMAIN;
+
+export const createClient = async (
+  userToken: string,
+  client: ClientCreate,
+): Promise<Client> => {
+  const token = await getAccessToken();
+  const userId = await getUserIdFromToken(userToken);
+
+  // Check if user already has a client
+  const userClientID = await getUserClientId(userId);
+
+  if (userClientID && userClientID.length > 0) {
+    return Promise.reject(
+      new ResponseError(400, "User already has a client", {} as Headers),
+    );
+  }
+
+  // Create new client
+  const newClient: Client = await parseResponse(
+    await fetch(`https://${authDomain}/api/v2/clients`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        ...client,
+        app_type: "non_interactive",
+      }),
+    }),
+  );
+
+  await setUserClientId(userId, newClient?.client_id || "");
+
+  return Promise.resolve(newClient);
+};
+
+export const getClient = async (userToken: string): Promise<Client> => {
+  const token: string = await getAccessToken();
+  const clientId = await getUserClientId(await getUserIdFromToken(userToken));
+
+  const client: Client = await parseResponse<Client>(
+    await fetch(
+      `https://${authDomain}/api/v2/clients/${encodeURIComponent(clientId)}`,
+      {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+      },
+    ),
+  );
+
+  return Promise.resolve(client);
```

Comment:
You can just return the client. You shouldn't need to Promise.resolve in an async function